site/blog/ecs-simplified.mdx

```diff
@@ -0,0 +1,124 @@
+SELECT iasql_install('aws_ecs_simplified');
+-- Create the load balancer, ECS service, task definition, etc. All ready afterwards for us to build and push to deploy the app.
+INSERT INTO ecs_simplified (app_name, app_port, image_tag, public_ip) VALUES ('<app_name>', 8088, 'latest', true);
+SELECT iasql_apply();
+-- Use the "high-level" ecr_build function to build the folder "examples/ecs-fargate/prisma/app" from "iasql-engine" Github repo and push it to ECR
+SELECT ecr_build(
+  'https://github.com/iasql/iasql-engine/', -- the Github repo URL
+  (SELECT id FROM repository WHERE repository_name = '<app_name>-repository')::varchar(255), -- ECR repo for the image to be pushed
+  './examples/ecs-fargate/prisma/app', -- the subdirectory in Github repo
+  'main', -- the Github branch or ref
+  '' -- Github personal access token - can be omitted if public repository
+);
+```
+```sql title="Create a security group named 'allow-all' attached to the VPC with ID 'vpc-f56cff91'. Then add a rule to it allowing all incoming traffic on port '80' from '0.0.0.0/0'"
+-- Install the aws_security_group IaSQL module
+SELECT iasql_install('aws_security_group');
+-- "Create a security group named `allow-all` attached to the VPC with ID `vpc-f56cff91`"
+INSERT INTO security_group (description, group_name, vpc_id)
+VALUES ('allows all incoming traffic on port 80', 'allow-all', (SELECT id FROM vpc WHERE vpc_id = 'vpc-08c0a68025c2983a2'));
+-- "Then add a rule to it allowing all incoming traffic on port `80` from `0.0.0.0/0`."
+INSERT INTO security_group_rule (is_egress, ip_protocol, from_port, to_port, cidr_ipv4, description, security_group_id)
+VALUES (false, 'tcp', 80, 80, '0.0.0.0/8', 'allow all to 80', (SELECT id FROM security_group WHERE group_name = 'allow-all'));
+```
+
+
+## ** An Example of `ecs_simplified` Module **
+Let's say we are going to deploy [this simple Express.js app](https://github.com/iasql/iasql-engine/tree/main/examples/ecs-fargate/prisma/app) to the ECS. It has a `Dockerfile` ready which `npm install`s the `express` package and then starts the Express server which listens on the port `8088`.
+
+Assuming that you have an IaSQL database, let's go and deploy the above app to our AWS account.
+```sql title="Deploy a simple Express.js app from a Github repository to ECS" showLineNumbers TheButton="Deploy Now"
+SELECT iasql_install('aws_ecs_simplified');
+INSERT INTO ecs_simplified(app_name, app_port, image_tag, public_ip) VALUES(
+  'simple-express',
+  8088,
+  'latest',
+  true
+);
+SELECT iasql_apply();
+
+SELECT ecr_build(
+  'https://github.com/iasql/iasql-engine/', -- the Github repo URL
+  (SELECT id
+    FROM repository
+    WHERE repository_name = 'simple-express-repository')::varchar(255), -- ECR repo for the image to be pushed
+  './examples/ecs-fargate/prisma/app', -- the subdirectory in Github repo
+  'main', -- the Github branch or ref
+  '' -- Github personal access token - can be omitted if public repository
```

Comment:
Now that it's the last argument, can we write `NULL` instead of an empty string, or does that still fail?

Reply:
It works with `NULL`. I've changed all occurrences to use `NULL` argument.